src/domain/transfer/commands/index.js

```diff
@@ -90,22 +130,60 @@ const prepareExecute = (payload, done) => {
         }
-        var notificatioMsgForCredits = {
+        var notificationMsgForCredits = {
           from: transfer.debits[0].account,
           to: transfer.credits[0].account,
           payload: result
         }
 
-        const topicForDebitNotifications = Kafka.tansformAccountToPrepareNotificationTopicName(UrlParser.nameFromAccountUri(notificatioMsgForDebits.to))
-        const topicForCreditNotifications = Kafka.tansformAccountToPrepareNotificationTopicName(UrlParser.nameFromAccountUri(notificatioMsgForCredits.to))
-        // TODO: WS Notifications to be re-worked so that it sends a notification to each DFSP
-        return Kafka.Producer.send({topic: topicForDebitNotifications, key: id, message: JSON.stringify(notificatioMsgForDebits)}).then(result => {
-        // return Kafka.send(topicForNotifications, id, result).then(result => {
+        const topicForDebitNotifications = Kafka.tansformAccountToPrepareNotificationTopicName(UrlParser.nameFromAccountUri(notificationMsgForDebits.to))
+        const topicForCreditNotifications = Kafka.tansformAccountToPrepareNotificationTopicName(UrlParser.nameFromAccountUri(notificationMsgForCredits.to))
+        /* const debitNotificationPromise = new Promise(function (resolve, reject) {
+          return Kafka.Producer.send({
+            topic: topicForDebitNotifications,
+            key: id,
+            message: JSON.stringify(notificationMsgForDebits)
+          }).then(result => {
+            // return Kafka.send(topicForNotifications, id, result).then(result => {
+            if (result) {
+              done()
+              Logger.info('result debit true')
+              return resolve(true)
+            } else {
+              done()
+              Logger.info('result debit false')
+              return reject(false)
+            }
+          }).catch(reason => {
+            Logger.error(`Transfers.prepareExecute.prepare.debit:: ERROR:'${reason}'`)
+            done() // TODO: Need to handle errors for Prepare Execution process
+            return reject(reason)
+          })
+        })
+        const creditNotificationPromise = new Promise(function (resolve, reject) {
+          return Kafka.Producer.send({
+            topic: topicForCreditNotifications,
+            key: id,
+            message: JSON.stringify(notificationMsgForCredits)
+          }).then(result => {
+            // return Kafka.send(topicForNotifications, id, result).then(result => {
+            if (result) {
+              done()
+              Logger.info('result credit true')
+              return resolve(true)
+            } else {
+              done()
+              Logger.info('result credit false')
+              return reject(false)
+            }
+          })
+        }) */
+        return Promise.all([debitNotificationPromise(topicForDebitNotifications, notificationMsgForDebits, id), creditNotificationPromise(topicForCreditNotifications, notificationMsgForCredits, id)]).then(result => {
```

Comment:
I see no difference between the debitNotificationPromise and the creditNotificationPromise methods except for the error message. Rather use a single method and add the error message as an input.